sdk/src/logs/logger_provider_factory.cc

```diff
@@ -12,42 +14,42 @@ namespace sdk
 namespace logs
 {
 
-std::unique_ptr<opentelemetry::logs::LoggerProvider> LoggerProviderFactory::Create(
+std::unique_ptr<opentelemetry::sdk::logs::LoggerProvider> LoggerProviderFactory::Create(
```

Comment:
Though returning the `sdk::\<signal\>provider` is the right approach, not sure if we should introduce the API-breaking changes in SDK without deprecating, specifically as it may introduce code change in the application. Probably finding the way to deprecate, and keep the same method name `Create`.

Reply:
Implemented a deprecation flag, so that users can adjust the code at their own pace.

I think we should remove this flag very soon, by 1.17.0, otherwise the codebase will rot with too many ifdef.

Providing an easier migration path to avoid forcing code changes overnight is one thing, but this is still a **_migration path_**, as the old code is meant to be removed and can not stay forever.